services/doctor/fix16961.go

```diff
@@ -249,6 +249,8 @@ func fixBrokenRepoUnit16961(repoUnit *repo_model.RepoUnit, bs []byte) (fixed boo
 		if fixed, err := fixIssuesConfig16961(bs, cfg); !fixed {
 			return false, err
 		}
+	case unit.TypeActions:
+		// action unit is newly added, so skip here
```

Comment:
Maybe next the panic would be caused by "Packages"

![image](https://github.com/go-gitea/gitea/assets/2114189/13d4a4e0-4115-48a9-abe3-8ee68035a3b4)

Reply:
I do not think the logic still make sense nowadays.

At least, it could first try to decode the JSON, if fails, then `repoUnit.Type`, and keep the old logic.

Because new units shouldn't/couldn't break, and never need to be fixed.